Homework 14

annavladimab opened the pull request:
Hello, I've added the home task 14  related to Pytest 
Thank you!

annavladimab committed:
feat: hw 15

annavladimab committed:
feat: hw 14

AnastasiiaGor commented on Homeworks/homework_14/categorypage.py:
```diff
@@ -0,0 +1,146 @@
+
+        cart_product_text = cart_product_element.text
+        assert expected_cart_product_name.strip() == cart_product_text.strip(), f"Expected text: '{expected_cart_product_name}', Actual text: '{cart_product_text}'"
+        time.sleep(110)
```
it's better avoid using sleep and use waiting for a specific event instead
also in this case waiting for 110sec looks unnecesarily long

annavladimab replied:
time.sleep(110) has been replaced with Wait Element 
the only place where I had to leave it (5 seconds) is when the alert is executed
for other cases, I added a wait for the element or page loading, which helped solve the problem
Thank you!

AnastasiiaGor commented on Homeworks/homework_14/test_site.py:
```diff
@@ -0,0 +1,76 @@
+from fixture import driver_init, login_data, setup_class
+
+
+class BaseTest:
```
Test functions should correspond test scenarios so there should be two test functions in test class/module

annavladimab replied:
Thank you very much for this comment
2 classes have been defined and each has test functions in accordance with the task
class TestLogin and class TestCategory

annavladimab committed:
update: hw 14
1.Two classes were added TestLogin and TestCategory
2.Time was replaced with Wait Element

AnastasiiaGor commented on Homeworks/homework_14/test_site.py:
```diff
@@ -0,0 +1,52 @@
+class TestLogin:
+
+    @pytest.fixture(scope="function", autouse=True)
+    def test_login_page_displayed(self):
```
These two test functions represent single test scenario so there might be only one test function containing actions from these functions sequentially

annavladimab replied:
one test function containing actions was created \
Thank you!

AnastasiiaGor commented on Homeworks/homework_14/test_site.py:
```diff
@@ -0,0 +1,52 @@
+        self.login_page.click_on_login_btn()
+
+    def test_product_name_and_price(self, login_data):
+        self.pre_req(login_data)
```
It would be better to use fixture to implement test precondition (or teardown) instead of using separate test function pre_req. The second approach causes inconsistency because test function is expected to be a representation of a specific test scenario and not other entities.

annavladimab replied:
pre_req fixture was created for TestCategory 
Thank you!

annavladimab committed:
update: hw 14
1.one test function containing actions
2.fixture pre-req was created
